test/nl/surf/eduhub/validator/service/jobs/client_test.clj

```diff
@@ -0,0 +1,81 @@
+(ns nl.surf.eduhub.validator.service.jobs.client-test
+  (:require [babashka.http-client :as http]
+            [babashka.json :as json]
+            [clojure.string :as str]
+            [clojure.test :refer [deftest is testing]]
+            [environ.core :refer [env]]
+            [goose.client :as c]
+            [nl.jomco.http-status-codes :as http-status]
+            [nl.surf.eduhub.validator.service.api :as api]
+            [nl.surf.eduhub.validator.service.config :as config]
+            [nl.surf.eduhub.validator.service.config-test :as config-test]
+            [nl.surf.eduhub.validator.service.test-helper :as test-helper]))
+
+(def test-config
+  (first (config/load-config-from-env (merge config-test/default-env env))))
+
+(def app (api/compose-app test-config false))
+
+(defn- make-status-call [uuid]
+  (-> (app {:uri (str "/status/" uuid) :request-method :get})
+      (update :body json/read-str)
+      (select-keys [:status :body])))
+
+(defn- pop-queue [atm]
+  (let [old-val @atm]
+    (when-not (empty? old-val)
+      (let [item    (peek old-val)
+            new-val (pop old-val)]
+        (if (compare-and-set! atm old-val new-val)
+          item
+          (pop-queue atm))))))
+
+(defn- assert-ok-return-body [resp]
```

Comment:
seems more logical to do the assertion in the `make-status-call` function, and remove this function